fix: 회원 가입 시 response DTO 형태 변경 MemberDto -> MemberInfoResponseDto

## 문제
회원 가입(`/join`) 할 때 응답 값 오류

- 기존 응답 형태
```
Error: response status is 500Response bodyDownloadcan't parse JSON.  Raw result:  {"status":200,"message":"성공","data"}{"status":500,"message":"서버에 문제가 발생하였습니다.","data":[]}
```

## 분석
- 에러 코드
```
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class com.capstone.member.dto.resonse.MemberInfoResponseDto and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: com.capstone.common.dto.BasicResponse["data"])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1306) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:408) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.failForEmpty(UnknownSerializer.java:53) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.serialize(UnknownSerializer.java:30) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:733) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:774) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1572) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:1061) ~[jackson-databind-2.14.2.jar!/:2.14.2]
	at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.writeInternal(AbstractJackson2HttpMessageConverter.java:483) ~[spring-web-6.0.8.jar!/:6.0.8]
	... 84 common frames omitted
```

## 해결
응답 값을 DTO 형으로 넘기지 않고, 사용자가 가입한 `username`을 넘김

- 변경된 응답 형태
```
{
  "status": 200,
  "message": "성공",
  "data": "siyeonSon"  // 사용자의 username
}
```

## ETC
XxResponseDto 형태가 Spring Security의 Filter에서 형 변환 문제가 발생하는 것 같아요.
정확히는 `JwtTokenFilter`에서 `doFilter()` 할 때 문제가 발생해요.

BasicResponse 수정 필요해 보여요.
응답 형태가 모두 변경될 것으로 예상됩니다.
추후 프론트엔드 팀과 회의 후 전반적인 응답값 형태 변화를 고려해 보아도 좋을 것 같아요!

--- a/bbiyong-api/src/main/java/com/capstone/member/controller/AuthController.java
+++ b/bbiyong-api/src/main/java/com/capstone/member/controller/AuthController.java
@@ -3,7 +3,6 @@ package com.capstone.member.controller;
 import com.capstone.common.dto.BasicResponse;
 import com.capstone.member.dto.request.JoinRequestDto;
 import com.capstone.member.dto.request.LoginRequestDto;
-import com.capstone.member.dto.resonse.MemberInfoResponseDto;
 import com.capstone.member.service.AuthService;
 import io.swagger.v3.oas.annotations.Operation;
 import io.swagger.v3.oas.annotations.tags.Tag;
@@ -24,12 +23,8 @@ public class AuthController {
     @PostMapping("/join")
     @Operation(summary = "회원 가입")
     public ResponseEntity<BasicResponse> join(@RequestBody JoinRequestDto joinRequestDto) {
-        var memberDto = authService.join(joinRequestDto.getUsername(), joinRequestDto.getPassword());
-        return basicResponse.ok(
-                MemberInfoResponseDto.builder()
-                        .userId(memberDto.getId())
-                        .username(memberDto.getUsername())
-                        .build());
+        var memberInfoResponseDto = authService.join(joinRequestDto.getUsername(), joinRequestDto.getPassword());
+        return basicResponse.ok(memberInfoResponseDto.getUsername());
     }
 
     @PostMapping("/login")

--- a/bbiyong-api/src/main/java/com/capstone/member/dto/resonse/MemberInfoResponseDto.java
+++ b/bbiyong-api/src/main/java/com/capstone/member/dto/resonse/MemberInfoResponseDto.java
@@ -1,8 +1,11 @@
 package com.capstone.member.dto.resonse;
 
+import com.capstone.member.domain.Member;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
+import lombok.Getter;
 
+@Getter
 @Builder
 @AllArgsConstructor
 public class MemberInfoResponseDto {
@@ -10,5 +13,12 @@ public class MemberInfoResponseDto {
     private final Long userId;
 
     private final String username;
+
+    public static MemberInfoResponseDto of(Member member) {
+        return MemberInfoResponseDto.builder()
+                .userId(member.getId())
+                .username(member.getUsername())
+                .build();
+    }
 
 }

--- a/bbiyong-api/src/main/java/com/capstone/member/service/AuthService.java
+++ b/bbiyong-api/src/main/java/com/capstone/member/service/AuthService.java
@@ -3,7 +3,7 @@ package com.capstone.member.service;
 import com.capstone.common.exception.BusinessException;
 import com.capstone.common.exception.ErrorCode;
 import com.capstone.member.domain.Member;
-import com.capstone.member.dto.MemberDto;
+import com.capstone.member.dto.resonse.MemberInfoResponseDto;
 import com.capstone.member.repository.MemberRepository;
 import com.capstone.security.service.CustomUserService;
 import com.capstone.security.util.JwtTokenUtils;
@@ -28,7 +28,7 @@ public class AuthService {
     private Long expiredTimeMs;
 
     @Transactional
-    public MemberDto join(String username, String password) {
+    public MemberInfoResponseDto join(String username, String password) {
         validateUnique(username);
 
         var member = Member.builder()
@@ -38,7 +38,7 @@ public class AuthService {
 
         var savedMember = memberRepository.save(member);
 
-        return MemberDto.of(savedMember);
+        return MemberInfoResponseDto.of(savedMember);
     }
 
     public String login(String username, String password) {

